usd/src/usd_parser/USDLinks.cc

```diff
@@ -0,0 +1,162 @@
+
+#include "sdf/usd/usd_parser/USDTransforms.hh"
+
+namespace sdf
+{
+// Inline bracket to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+//
+namespace usd
+{
+void GetInertial(
+  const pxr::UsdPrim &_prim,
+  ignition::math::Inertiald &_inertial)
+{
+  float mass;
+  pxr::GfVec3f centerOfMass;
+  pxr::GfVec3f diagonalInertia;
+  pxr::GfQuatf principalAxes;
+
+  ignition::math::MassMatrix3d massMatrix;
+
+  if (_prim.HasAPI<pxr::UsdPhysicsRigidBodyAPI>())
+  {
+    if (pxr::UsdAttribute massAttribute =
+      _prim.GetAttribute(pxr::TfToken("physics:mass")))
+    {
+      massAttribute.Get(&mass);
+
+      if (pxr::UsdAttribute centerOfMassAttribute =
+        _prim.GetAttribute(pxr::TfToken("physics:centerOfMass")))
+      {
+        centerOfMassAttribute.Get(&centerOfMass);
+      }
+
+      if (pxr::UsdAttribute diagonalInertiaAttribute =
+        _prim.GetAttribute(pxr::TfToken("physics:diagonalInertia")))
+      {
+        diagonalInertiaAttribute.Get(&diagonalInertia);
+      }
+
+      if (pxr::UsdAttribute principalAxesAttribute =
+        _prim.GetAttribute(pxr::TfToken("physics:principalAxes"))) {
+        principalAxesAttribute.Get(&principalAxes);
+      }
+
+      // Added a diagonal inertia to avoid crash with the physics engine
+      if (diagonalInertia == pxr::GfVec3f(0, 0, 0))
+      {
+        diagonalInertia = pxr::GfVec3f(0.0001, 0.0001, 0.0001);
+      }
+
+      // Added a massto avoid crash with the physics engine
+      if (mass < 0.0001)
+      {
+        mass = 0.1;
+      }
+
+      massMatrix.SetMass(mass);
+      massMatrix.SetDiagonalMoments(
+        ignition::math::Vector3d(
+          diagonalInertia[0],
+          diagonalInertia[1],
+          diagonalInertia[2]));
+
+      _inertial.SetPose(ignition::math::Pose3d(
+          ignition::math::Vector3d(
+            centerOfMass[0], centerOfMass[1], centerOfMass[2]),
+          ignition::math::Quaterniond(1.0, 0, 0, 0)));
+
+      _inertial.SetMassMatrix(massMatrix);
+    }
+  }
+  else
+  {
+    auto parent = _prim.GetParent();
+    if (parent)
+    {
+      GetInertial(parent, _inertial);
+    }
+  }
+}
+
+void ParseUSDLinks(
+  const pxr::UsdPrim &_prim,
+  const std::string &_nameLink,
+  sdf::Link *_link,
+  USDData &_usdData)
+{
+  std::string primNameStr = _prim.GetPath().GetName();
+  std::string primPathStr = pxr::TfStringify(_prim.GetPath());
+  std::string primType = _prim.GetPrimTypeInfo().GetTypeName().GetText();
+
+  std::pair<std::string, std::shared_ptr<USDStage>> data =
+    _usdData.FindStage(primNameStr);
+
+  // Is this a new link ?
+  if (_link == nullptr)
+  {
+    _link = new sdf::Link();
+    _link->SetName(ignition::common::basename(_nameLink));
+
+    ignition::math::Pose3d pose;
+    ignition::math::Vector3d scale(1, 1, 1);
+    GetTransform(_prim, _usdData, pose, scale, "");
+    size_t nSlash = std::count(_nameLink.begin(), _nameLink.end(), '/');
+    if (nSlash > 1)
+      _link->SetRawPose(pose);
```

Comment:
@ahcorde I left a [previous comment](https://github.com/ignitionrobotics/sdformat/pull/902#discussion_r840922946) about this, but I still don't understand the purpose of this code. Can you explain why we need to count the slashes and make sure it's `> 1` before setting the pose? I think it would be nice to leave a comment in the code in case this has to be re-visited later.

Reply:
Added a explanation and added TODO